src/index.js

```diff
@@ -36,14 +37,12 @@ export default function (babel) {
 
           let src = template.split('\n');
 
-          const minIndent = src.reduce(
-            (minIndent, line) => {
-              return line.trim().length ? Math.min(/^ */.exec(line)[0].length, minIndent) : minIndent;
-            },
-            Infinity
-          );
+          const minIndent = common(
+            src.filter(indent => indent.trim() !== '')
+            .map(line => /^[ \t]*/.exec(line)[0])
+          )
 
-          src = src.map(line => line.substr(minIndent)).join('\n');
+          src = src.map(line => line.replace(minIndent, '')).join('\n');
```

Comment:
I'd prefer to `substr(minIndent.length)` here.

Reply:
Out of curiosity, do you have any particular reason for the preference?

Reply:
I don't like depending on the rather odd behaviour of `replace` that when given a string, it actually only replaces the first occurrence. As it stands, it's always going to take a while to mentally confirm that having something like the following won't break:

```js
const elmenets = pug`
  div(foo="bar"  bar="foo")
`;
```

i.e. it's only because `replace` is actually "replace first occurrence" that it doesn't remove the space between the two attributes.

Reply:
That makes sense, I'm used to this behavior but I can see how it would be confusing.